vignettes/vizualizing_demography.Rmd

```diff
@@ -0,0 +1,349 @@
+
+This table illustrates the general structure of the Demographic Transition Model, and on the left hand side axis there are suggested values for what can be considered "high" and "low" birth and death rates ("high" being approximately >=40 per 1,000, "low" being approximately <= 10 per 1,000). While the rate of change in the rates can vary, the overall trend tends to be that deathrates fall first, then birth rates, and this results in a gradual population increase. A detailed explanation of this can be found at [Our World in Data.](https://ourworldindata.org/world-population-growth)
+
+Most countries were in Stage 1 pre industrial revolution, and many of what are now called high income countries have experienced the falling and then stabilizing death and birth rates that characterize the progression through the stages. The rates fall often as a result of the stronger infrastructure, healthcare, and education that come through increased country net income. Countries can be categorized by stage in their demographic transition, often by looking at the rate of change of these rates relative to each other over time.
+
+```{r data processing}
+cdr <- peeps::cdr
+cbr <- peeps::cbr
+pop <- peeps::pop
+
+cdr$value<-cdr$value*1000
+cbr$value<-cbr$value*1000
+
+names(cdr)[7]<-"deathrate"
+names(cbr)[7]<-"birthrate"
+names(pop)[7]<-"population"
+
+rates<-merge(cdr, cbr)
+rates <- subset(rates, select=-c(age_from, age_to))
+pop<- subset(pop, select=-c(age_from, age_to))
+all_data<-merge(rates, pop)
+
+```
+
+We can plot the crude birth rates, crude death rates, and total population count over time to visualize the observed and predicted demographic shifts for the countries in our dataset. 
+
+```{r dtm plots, eval=FALSE}
+# Making an animation of the demography plots
+makeplot <- function(){
+  datalist <- split(all_data, all_data$country_code)
+  current_year <-format(Sys.Date(), "%Y")
+  lapply(datalist, function(data){
+    coeff <- max(data$population)/max(data$deathrate) # get scaling coefficient 
+    p <- ggplot2::ggplot(data, aes(x=year)) +
+    geom_line(aes(y=birthrate, color="Crude Birth Rate")) +
+    geom_line(aes(y=deathrate, color="Crude Death Rate")) +
+    geom_line(aes(y=population/coeff, color="Population")) +
+    geom_vline(xintercept=as.numeric(current_year),linetype=2) +
+    geom_text(aes(x=as.numeric(current_year)+10, label=current_year, y=0)) +
+    scale_y_continuous(
+      
+      # Features of the first axis
+      name = "Average rate per 1,000",
+      
+      # Add a second axis and specify its features
+      sec.axis = sec_axis(~.*coeff, name="Population")
+    ) +
+    labs(x="Year", title=paste0(data$country_code, " Demographic Plot"), col="Values") +
+    scale_color_manual(values = c(`Crude Birth Rate` = "green", `Crude Death Rate`= "purple", `Population` = "blue")) +
+    theme_bw()
+    print(p)
+  })
+}
+gif_file <- file.path('dtm.gif')
+gifski::save_gif(makeplot(), gif_file, 1280, 720, res = 144)
```

Comment:
We don't want to be saving things when building vignettes (these are build on users machines when installing). I think it would be better to remove the gifski depedency, pre-make the gifs in a script stored in data-raw/ and then just load the gif into the vignette.

Reply:
Exactly what I thought! So I built them beforehand - and you'll see that eval=FALSE and I just included the code so that people have prewritten code the can tinker with to make the demography plots. Can easily move that code elsewhere, though, just let me know!